include/veriblock/storage/block_repository.hpp

```diff
@@ -60,26 +62,25 @@ struct WriteBatch {
    * @param block to be written in a batch
    * @return true if we overwrite existing block, false otherwise.
    */
-  virtual bool put(const stored_block_t& block) = 0;
+  virtual void put(const stored_block_t& block) = 0;
 
   /**
    * Remove a single block from storage identified by its hash.
    * @param hash block hash
    * @return true if removed, false if no such element found.
    */
-  virtual bool removeByHash(const hash_t& hash) = 0;
+  virtual void removeByHash(const hash_t& hash) = 0;
 
   /**
-   * Remove potentially many blocks at given height.
-   * @param height block height
-   * @return number of blocks removed.
+   * Clear batch from any modifying operations.
    */
-  virtual size_t removeByHeight(height_t height) = 0;
+  virtual void clear() = 0;
 
   /**
-   * Clear batch from any modifying operations.
+   * Efficiently commit given batch on-disk. Clears batch from changes.
+   * @param repo
    */
-  virtual void clear() = 0;
+  virtual void commit(BlockRepository<Block>& repo) = 0;
```

Comment:
Shouldn't it be reverse way? A BlockRepository with commit(WriteBatch) method.

Reply:
hard to implement it this way... :-(

Reply:
because repo has access only to parent class "WriteBatch", and without implementation details it is hard